src/sites/twitch-twilight/modules/host_button.js

```diff
@@ -0,0 +1,336 @@
+				removeFromHosts: event => this.removeUserFromHosts(event),
+				setActiveTab: tab => {
+					this.vueEl.$children[0]._data.activeTab = this.activeTab = tab;
+				},
+				updatePopper: () => {
+					if (this._auto_host_tip) this._auto_host_tip.update();
+				},
+				updateCheckbox: e => {
+					const t = e.target,
+						setting = t.dataset.setting;
+					let state = t.checked;
+				
+					if ( setting === 'strategy' )
+						state = state ? 'random' : 'ordered';
+					else if ( setting === 'deprioritize_vodcast' )
+						state = ! state;
+				
+					this.updateAutoHostSetting(setting, state);
+				}
+			})
+		});
+
+		return this.vueEl.$el;
+	}
+
+	async fetchAutoHosts() {
+		const user = this.resolve('site').getUser();
+		if ( ! user )
+			return;
+
+		let data;
+		try {
+			data = await fetch('https://api.twitch.tv/kraken/autohost/list', {
+				headers: {
+					'Accept': 'application/vnd.twitchtv.v4+json',
+					'Authorization': `OAuth ${user.authToken}`
+				}
+			}).then(r => {
+				if ( r.ok )
+					return r.json();
+
+				throw r.status;
+			});
+
+		} catch(err) {
+			this.log.error('Error loading auto host list.', err);
+			return;
+		}
+
+		return this.autoHosts = data.targets;
+	}
+
+	async fetchAutoHostSettings() {
+		const user = this.resolve('site').getUser();
+		if ( ! user )
+			return;
+
+		let data;
+		try {
+			data = await fetch('https://api.twitch.tv/kraken/autohost/settings', {
+				headers: {
+					'Accept': 'application/vnd.twitchtv.v4+json',
+					'Authorization': `OAuth ${user.authToken}`
+				}
+			}).then(r => {
+				if ( r.ok )
+					return r.json();
+
+				throw r.status;
+			});
+
+		} catch(err) {
+			this.log.error('Error loading auto host settings.', err);
+			return;
+		}
+
+		return this.autoHostSettings = data.settings;
+	}
+
+	queueHostUpdate(newHosts) {
+		if (this._host_update_timer) clearTimeout(this._host_update_timer);
+		
+		this._host_update_timer = setTimeout(() => {
+			this._host_update_timer = undefined;
+			this.updateAutoHosts(newHosts);
+		}, 1000);
+	}
+	
+	rearrangeHosts(oldIndex, newIndex) {
+		const newHosts = this.autoHosts.slice(0);
+		
+		const host = newHosts.splice(oldIndex, 1)[0];
+		newHosts.splice(newIndex, 0, host);
+		
+		this.queueHostUpdate(newHosts);
+	}
+	
+	currentRoomInHosts() {
+		return this.getAutoHostIDs(this.autoHosts).includes(parseInt(this._current_channel_id, 10));
+	}
+
+	addCurrentRoomToHosts() {
+		const newHosts = this.autoHosts.slice(0);
+		newHosts.push({ _id: parseInt(this._current_channel_id, 10)});
+
+		this.updateAutoHosts(newHosts);
+	}
+
+	removeUserFromHosts(event) {
+		const id = event.target.closest('.ffz--host-user').getAttribute('data-id');
+		
+		const newHosts = [];
+		for (let i = 0; i < this.autoHosts.length; i++) {
+			if (`${this.autoHosts[i]._id}` !== id) newHosts.push(this.autoHosts[i]);
+		}
+
+		this.updateAutoHosts(newHosts);
+	}
+
+	getAutoHostIDs(hosts) { // eslint-disable-line
```

Comment:
When disabling eslint rules, please be specific about which rule you're disabling. `class-methods-use-this` this time, right?